File: app/src/main/java/com/example/productbtl/Adapter/receipt_Adapter.java
package com.example.productbtl.Adapter;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.productbtl.Object.receipt;
import com.example.productbtl.R;
import com.example.productbtl.myinterface.buttonClickListenner;

import java.util.List;

public class receipt_Adapter extends RecyclerView.Adapter<receipt_Adapter.receipt_Adapterholder>{

    private List<receipt> list;
    private buttonClickListenner listenner;

    public receipt_Adapter(List<receipt> list , buttonClickListenner listenner) {
        this.list = list;
        this.listenner = listenner;
    }

    public void setData(List<receipt> list ){
        this.list = list;
        notifyDataSetChanged();
    }
    @NonNull
    @Override
    public receipt_Adapterholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.receipt_items,parent,false);
        return new receipt_Adapterholder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull receipt_Adapterholder holder, int position) {

        receipt receipt= list.get(position);
        if(receipt == null)
            return;
        holder.tv_address.setText(receipt.getAddress());
        holder.tv_date.setText(receipt.getDate());
        holder.tv_name.setText(receipt.getName());
        holder.tv_id.setText(receipt.getId());
        holder.tv_phonenumber.setText(receipt.getPhonenumber());
        holder.tv_price.setText(receipt.getPrice());
        holder.tv_product.setText(receipt.getProduct());
        holder.tv_status.setText(receipt.getStatus());
        holder.btn_delete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listenner.onClickDeleteItem(receipt);
            }
        });
    }

    @Override
    public int getItemCount() {
        if(list != null)
            return list.size();
        else
            return 0;
    }

    public class receipt_Adapterholder extends RecyclerView.ViewHolder {
        private TextView tv_name , tv_date , tv_address,tv_id , tv_phonenumber , tv_status , tv_product , tv_price;
        private Button btn_delete;

        public receipt_Adapterholder(@NonNull View itemView) {
            super(itemView);
            tv_address = itemView.findViewById(R.id.address_of_receipt);
            tv_date = itemView.findViewById(R.id.date_of_receipt);
            tv_name = itemView.findViewById(R.id.name_of_receipt);
            tv_id = itemView.findViewById(R.id.id_of_receipt);
            tv_phonenumber = itemView.findViewById(R.id.phone_of_receipt);
            tv_price = itemView.findViewById(R.id.price_of_receipt);
            tv_product = itemView.findViewById(R.id.product_of_receipt);
            tv_status = itemView.findViewById(R.id.status_of_receipt);
            btn_delete = itemView.findViewById(R.id.btn_delete_receipt);

        }
    }
}

File: app/src/main/java/com/example/productbtl/Adapter/receipt_Adapter.kt
package com.example.productbtl.Adapter

import com.example.productbtl.Object.product
import com.example.productbtl.Object.cart
import com.example.productbtl.myinterface.imageClickListenner
import androidx.recyclerview.widget.RecyclerView
import com.example.productbtl.Adapter.cart_Adapter.cart_AdapterHolder
import android.view.ViewGroup
import android.view.LayoutInflater
import com.example.productbtl.R
import com.example.productbtl.Activity.MainActivity
import com.bumptech.glide.Glide
import com.example.productbtl.Adapter.menu_Adapter.clickItemMenu
import com.example.productbtl.Adapter.menu_Adapter.menu_AdapterHolder
import com.example.productbtl.Adapter.detail_Adapter.detail_AdapterHolder
import com.firebase.ui.database.FirebaseRecyclerOptions
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.example.productbtl.Adapter.product_Adapter.product_AdapterHolder
import androidx.appcompat.app.AppCompatActivity
import com.example.productbtl.Fragment.OrderFragment
import com.example.productbtl.Object.receipt
import com.example.productbtl.myinterface.buttonClickListenner
import com.example.productbtl.Adapter.receipt_Adapter.receipt_Adapterholder
import com.example.productbtl.Adapter.category_Adapter.categoryViewHolder
import com.example.productbtl.Object.category
import androidx.recyclerview.widget.LinearLayoutManager
import com.example.productbtl.Adapter.rcvCategory_Adapter
import com.example.productbtl.Object.slideshow
import androidx.viewpager.widget.PagerAdapter
import com.example.productbtl.Adapter.rcvCategory_Adapter.rcvCategory_AdapterHolder
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.example.productbtl.Fragment.HomeFragment
import com.example.productbtl.Fragment.ReceiptFragment
import com.example.productbtl.Fragment.ProfileFragment
import android.os.Bundle
import com.google.android.material.navigation.NavigationBarView
import com.example.productbtl.Fragment.ProductFragment
import com.example.productbtl.Fragment.CartFragment
import com.example.productbtl.Fragment.DetailCartFragment
import android.app.ProgressDialog
import com.google.firebase.auth.FirebaseAuth
import android.content.Intent
import android.view.View
import android.widget.*
import com.example.productbtl.Activity.RegesterActivity
import com.google.android.gms.tasks.OnCompleteListener
import com.google.firebase.auth.AuthResult
import com.example.productbtl.Activity.LoginActivity
import com.example.productbtl.Adapter.cart_Adapter
import androidx.viewpager.widget.ViewPager
import me.relex.circleindicator.CircleIndicator
import com.example.productbtl.Adapter.category_Adapter
import androidx.recyclerview.widget.GridLayoutManager
import com.example.productbtl.Adapter.menu_Adapter
import com.example.productbtl.Adapter.slideshow_Adapter
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.ValueEventListener
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.example.productbtl.Adapter.product_Adapter
import com.example.productbtl.Adapter.receipt_Adapter
import com.example.productbtl.Adapter.detail_Adapter

class receipt_Adapter(
    private var list: List<receipt?>?,
    private val listenner: buttonClickListenner
) : RecyclerView.Adapter<receipt_Adapterholder>() {
    fun setData(list: List<receipt?>?) {
        this.list = list
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): receipt_Adapterholder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.receipt_items, parent, false)
        return receipt_Adapterholder(view)
    }

    override fun onBindViewHolder(holder: receipt_Adapterholder, position: Int) {
        val receipt = list!![position] ?: return
        holder.tv_address.text = receipt.address
        holder.tv_date.text = receipt.date
        holder.tv_name.text = receipt.name
        holder.tv_id.text = receipt.id
        holder.tv_phonenumber.text = receipt.phonenumber
        holder.tv_price.text = receipt.price
        holder.tv_product.text = receipt.product
        holder.tv_status.text = receipt.status
        holder.btn_delete.setOnClickListener { listenner.onClickDeleteItem(receipt) }
    }

    override fun getItemCount(): Int {
        return if (list != null) list!!.size else 0
    }

    inner class receipt_Adapterholder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tv_name: TextView
        val tv_date: TextView
        val tv_address: TextView
        val tv_id: TextView
        val tv_phonenumber: TextView
        val tv_status: TextView
        val tv_product: TextView
        val tv_price: TextView
        val btn_delete: Button

        init {
            tv_address = itemView.findViewById(R.id.address_of_receipt)
            tv_date = itemView.findViewById(R.id.date_of_receipt)
            tv_name = itemView.findViewById(R.id.name_of_receipt)
            tv_id = itemView.findViewById(R.id.id_of_receipt)
            tv_phonenumber = itemView.findViewById(R.id.phone_of_receipt)
            tv_price = itemView.findViewById(R.id.price_of_receipt)
            tv_product = itemView.findViewById(R.id.product_of_receipt)
            tv_status = itemView.findViewById(R.id.status_of_receipt)
            btn_delete = itemView.findViewById(R.id.btn_delete_receipt)
        }
    }
}